rmgpy/molecule/molecule.py

```diff
@@ -2618,6 +2618,152 @@ def enumerate_bonds(self):
 
         return dict(bond_count)
 
+    def get_surface_sites(self):
+        """
+        Get a list of surface site atoms in the molecule.
+        Returns:
+            List(Atom): A list containing the surface site atoms in the molecule
+        """
+        cython.declare(atom=Atom)
+        return [atom for atom in self.atoms if atom.is_surface_site()]
+
+    def get_adatoms(self):
+        """
+        Get a list of adatoms in the molecule.
+        Returns:
+            List(Atom): A list containing the adatoms in the molecule
+        """
+        cython.declare(surface_site=Atom, adatoms=list)
+        adatoms = []
+        for surface_site in self.get_surface_sites():
+            if surface_site.bonds:
+                adatoms.append(list(surface_site.bonds.keys())[0])
+        return adatoms
+
+    def get_desorbed_molecules(self):
+        """
+        Get a list of desorbed molecules by desorbing the molecule from the surface.
+        
+        Returns a list of Molecules.  Each molecule's atoms will be labeled corresponding to
+        the bond order with the surface:
+        '*1' - Single bond
+        '*2' - double bond
+        '*3' - triple bond
+        '*4' - quadruple bond
+        """
+        if not self.contains_surface_site():
+            return []
+
+        dummy_molecule = self.copy(deep=True)
+        dummy_molecule.clear_labeled_atoms()
+        sites_to_remove = dummy_molecule.get_surface_sites()
+        adsorbed_atoms = []
+        for site in sites_to_remove:
+            numbonds = len(site.bonds)
+            if numbonds == 0:
+                # vanDerWaals
+                pass
+            else:
+                assert len(site.bonds) == 1, "Each surface site can only be bonded to 1 atom"
+                bonded_atom = list(site.bonds.keys())[0]
```

Comment:
I don't know how Cython will interact with this, but in pure Python this is slightly better to get the first key:
```suggestion
                bonded_atom = next(iter(site.bonds))
```

Reply:
Just realized you can get the bonded atom and the bond at the same time with this, as we know there is only one entry in the dictionary:
```suggestion
                (bonded_atom, bond), = site.bonds.items()
```
(that trailing comma is crucial for the unpacking to work)
Then you don't need the line two below this.